service-base/src/main/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecks.java

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.plan;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import org.eclipse.hono.service.metric.MicrometerBasedMetrics;
+import org.eclipse.hono.util.TenantObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link ResourceLimitChecks} which uses metrics data from the prometheus backend to check if
+ * further connections or messages are allowed by comparing with the configured limits.
+ */
+@Component
```

Comment:
Is there a reason why you want to mark this as a component (but do not mark the constructor args as `@Autowired`)?
The configuration explicitly creates an instance using the constructor ...

Reply:
@sophokles73 `@Component` is not needed anymore. I will remove it.